src/backend/parser/parse_expr.c

```diff
@@ -1834,6 +1835,15 @@ transformCaseExpr(ParseState *pstate, CaseExpr *c)
 
 	newc->location = c->location;
 
+	/* Following hook will be used to set the typmod of all the CASE Branches. */
+	if(sql_dialect == SQL_DIALECT_TSQL && set_common_typmod_case_expr_hook)
+	{
+		/* calculating common_typemod for case expr */
+		int32 typmod = select_common_typmod(pstate, resultexprs, newc->casetype);
```

Comment:
we should move this under our hook to ensure minimal chances of merge conflict on engine side

Reply:
Resolved